src/app/work-item/work-item-list/work-item-detail/work-item-link/work-item-link.component.ts

```diff
@@ -90,98 +85,54 @@ export class WorkItemLinkComponent implements OnInit, OnChanges {
     createLink(event: any = null): void {
-         this.createLinkObject(this.workItem['id'], this.selectedWorkItem['id'], this.selectedLinkType['id']);
-         const tempValue = {'data': this.link};
-         this.WorkItemLinkService
-        .createLink(tempValue)
-        .then(response => {
-          this.workitemLinks.push(response);
-          this.calculateTotal();
-        }).catch((error) => {
-            console.log(error);
-        });
+        this.createLinkObject(
+           this.workItem['id'], 
+           this.selectedWorkItem['id'], 
+           this.selectedLinkType.linkId, 
+           this.selectedLinkType.linkType
+          );
+        const tempValue = {'data': this.link};
+        this.workItemService
+        .createLink(tempValue, this.workItem['id'])
+        .then((response: any) => {
+          this.selectedLinkType = false;
+          this.selectedWorkItem = null;
+        })
+        .catch ((error: any) => console.log(error));
     }
 
     // deleteLink(link : Link){
-    deleteLink(link : any): void{
-         this.WorkItemLinkService
-        .deleteLink(link)
-        .then(response => {
-          this.workitemLinks.forEach((item, index) => {
-              if (item['id'] === link['id']) {
-                  this.workitemLinks.splice(index, 1);
-                  this.calculateTotal();
-              }
-          });
-        }).catch((error) => {
-            console.log(error);
-        });
+    deleteLink(event: any, link : any, currentWorkItem: WorkItem): void{
+      event.stopPropagation();
+      this.workItemService
+      .deleteLink(link, currentWorkItem.id)
+      .then((response: any) => {
+        if (!this.workItem.relationalData.totalLinkCount) {
+          this.showLinkView = false;
+        }
+      })
+      .catch ((error: any) => console.log(error));
     }
 
     loadLinkTypes(): void {
-        this.WorkItemLinkService
-        .getLinkTypes()
-        .then((lTypes) => {
-            this.linkTypes = lTypes;
-            this.selectedLinkType = lTypes[0];
-            this.searchWorkItem.searchType = lTypes[0].relationships.target_type.data.id;
-        }).catch(() => {
-            console.log('Error in loading Link Types');
-        });
-    }
-    
-    loadAllLinks(): void{
-        this.WorkItemLinkService
-        .getAllLinks()
-        .then((links) => {
-            this.workitemLinks = links;
-        }).catch(() => {
-            console.log('Error in loading Links');
-        });
-
-        // This is a fast fix
-        // This logic is going to be changed 
-        // once we have associated links as relation for a work item
         this.workItemService
-        .getLocallySavedWorkItems()
-        .then((wItems) => {
-            let workItems = cloneDeep(wItems);
-            workItems.forEach((item: any) => {
-                this.workItemsMap[item.id] = item;
-            });
-            this.calculateTotal();
-        });
-    }
-
-    calculateTotal(): void{
```

Comment:
Because all these data manipulation is happening in the workItemService. 
Check models/work-item.ts file.
export class RelationalData {
  creator?: User;
  comments?: Comment[];
  assignees?: User[];
  linkDicts?: LinkDict[];
  totalLinkCount?: number;
}